src/app.ts

```diff
@@ -1,6 +1,87 @@
 
 const startApp = async () => {
+    const tbody = document.querySelector('[data-sink]')
+    const tr = tbody?.querySelectorAll('tr')
+    const prevBtn = document.querySelector('[data-prevbtn')
+    const nextBtn = document.querySelector('[data-nextbtn')
+    let url = 'https://randomapi.com/api/8csrgnjw?key=LEIX-GF3O-AG7I-6J84&page=1'
+    let response = await fetch(url)
+    let currentIndex = 1
+    let dataStore = {}
+    if(response.ok){
+        let data = await response.json();
+        dataStore = data.results[0]
+        insertDataRow(tr, dataStore, currentIndex)
+        disableBtn(dataStore.paging, prevBtn, nextBtn)
+
+    }
     
+    nextBtn?.addEventListener('click', async (e) => {
+        currentIndex++
+        if(dataStore[currentIndex]){
+            insertDataRow(tr, dataStore, currentIndex)
+        } else{
+            if(dataStore.paging.next){
+                let response = await fetch(dataStore.paging.next)
+                nextBtn.disabled = true // Disabled previous button on request to server
+                nextBtn.textContent  = "loading..."
+                if(response.ok){
+                    const data = await response.json();
+                    dataStore = data.results[0]
+                    insertDataRow(tr, dataStore, currentIndex)
+                    disableBtn(dataStore.paging, prevBtn, nextBtn)
+                    nextBtn.disabled = false  // Enable button after processing
+                    nextBtn.textContent = 'Next'
+
+                }
+            }
+        }
+    })
+
+    prevBtn?.addEventListener('click', async () => {
+        currentIndex--
+        if(dataStore[currentIndex]){
+            insertDataRow(tr, dataStore, currentIndex)
+        } else{
+            if(dataStore.paging.previous){
+                let response = await fetch(dataStore.paging.previous)
+                prevBtn.disabled = true // Disabled button on request to server
```

Comment:
Codacy has a fix for the issue: [Replace `················prevBtn.disabled·=·true` with `········prevBtn.disabled·=·true;`](https://app.codacy.com/gh/alic-xc/Test/pullRequest?prid=9965044)
```suggestion
        prevBtn.disabled = true; // Disabled button on request to server
```